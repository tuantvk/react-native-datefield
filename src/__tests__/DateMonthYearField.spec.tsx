import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DateMonthYearField from '../DateMonthYearField';

describe('DateMonthYearField component', () => {
  it('should match snapshot', () => {
    const component = shallow(<DateMonthYearField />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  describe('Props', () => {
    it('testID', () => {
      const component = shallow(<DateMonthYearField testID="date_input" />);
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('containerStyle', () => {
      const component = shallow(
        <DateMonthYearField containerStyle={{ marginVertical: 20 }} />
      );
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('styleInput', () => {
      const component = shallow(
        <DateMonthYearField styleInput={{ fontSize: 15 }} />
      );
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('labelDate', () => {
      const component = shallow(<DateMonthYearField labelDate="Input date" />);
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('labelMonth', () => {
      const component = shallow(
        <DateMonthYearField labelMonth="Input month" />
      );
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('labelYear', () => {
      const component = shallow(<DateMonthYearField labelYear="Input year" />);
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('defaultValue', () => {
      const component = shallow(
        <DateMonthYearField defaultValue={new Date('2021-03-25')} />
      );
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('onSubmit', () => {
      const component = shallow(<DateMonthYearField onSubmit={() => {}} />);
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('editable', () => {
      const component = shallow(<DateMonthYearField editable={false} />);
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('hideDate', () => {
      const component = shallow(<DateMonthYearField hideDate />);
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('hideDate - check total length TextInput', () => {
      const component = shallow(<DateMonthYearField hideDate />);
      expect(component.find('Memo(ForwardRef)')).toHaveLength(2);
    });

    it('placeholderTextColor', () => {
      const component = shallow(
        <DateMonthYearField placeholderTextColor="#000000" />
      );
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('maximumDate', () => {
      const component = shallow(
        <DateMonthYearField maximumDate={new Date(2023, 3, 12)} />
      );
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('minimumDate', () => {
      const component = shallow(
        <DateMonthYearField minimumDate={new Date(2020, 3, 12)} />
      );
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('handleErrors', () => {
      const component = shallow(<DateMonthYearField handleErrors={() => {}} />);
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
