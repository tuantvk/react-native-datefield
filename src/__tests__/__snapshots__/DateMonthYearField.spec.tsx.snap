// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateMonthYearField component Props containerStyle 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Object {
        "marginVertical": 20,
      },
    ]
  }
>
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Date"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Month"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={4}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Year"
    returnKeyType="done"
    value=""
  />
</View>
`;

exports[`DateMonthYearField component Props defaultValue 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Date"
    returnKeyType="next"
    value="25"
  />
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Month"
    returnKeyType="next"
    value="3"
  />
  <ForwardRef
    editable={true}
    maxLength={4}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Year"
    returnKeyType="done"
    value="2021"
  />
</View>
`;

exports[`DateMonthYearField component Props editable 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <ForwardRef
    editable={false}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Date"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={false}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Month"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={false}
    maxLength={4}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Year"
    returnKeyType="done"
    value=""
  />
</View>
`;

exports[`DateMonthYearField component Props labelDate 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Input date"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Month"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={4}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Year"
    returnKeyType="done"
    value=""
  />
</View>
`;

exports[`DateMonthYearField component Props labelMonth 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Date"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Input month"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={4}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Year"
    returnKeyType="done"
    value=""
  />
</View>
`;

exports[`DateMonthYearField component Props labelYear 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Date"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Month"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={4}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Input year"
    returnKeyType="done"
    value=""
  />
</View>
`;

exports[`DateMonthYearField component Props onSubmit 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Date"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Month"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={4}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Year"
    returnKeyType="done"
    value=""
  />
</View>
`;

exports[`DateMonthYearField component Props placeholderTextColor 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Date"
    placeholderTextColor="#000000"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Month"
    placeholderTextColor="#000000"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={4}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Year"
    placeholderTextColor="#000000"
    returnKeyType="done"
    value=""
  />
</View>
`;

exports[`DateMonthYearField component Props styleInput 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Date"
    returnKeyType="next"
    style={
      Object {
        "fontSize": 15,
      }
    }
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Month"
    returnKeyType="next"
    style={
      Object {
        "fontSize": 15,
      }
    }
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={4}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Year"
    returnKeyType="done"
    style={
      Object {
        "fontSize": 15,
      }
    }
    value=""
  />
</View>
`;

exports[`DateMonthYearField component Props testID 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
  testID="date_input"
>
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Date"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Month"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={4}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Year"
    returnKeyType="done"
    value=""
  />
</View>
`;

exports[`DateMonthYearField component should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Date"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={2}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Month"
    returnKeyType="next"
    value=""
  />
  <ForwardRef
    editable={true}
    maxLength={4}
    onBlur={[Function]}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Year"
    returnKeyType="done"
    value=""
  />
</View>
`;
